### PR TITLE
browserify-shim window.marked

### PR DESCRIPTION
**Pull request by Bjvanminnen**
This comes out of a piece of feedback from https://github.com/code-dot-org/code-dot-org/pull/7507

Right now apps depends on dashboard putting window.marked in the global namespace. Furthermore, in tests is wasn't actually defined, so we had a bunch of places in code where we would check for the presence of `window.marked` only because of tests.

This PR first cleans up a bunch of external globals, and sticks them in a single function that any test file can call. Then, instead of using window.marked, we shim it using browserify-shim and do `require('marked');`. Arguably, we could/should do the same thing for other globals like jQuery and React, though that's going to touch a lot more files.

Assigning to @Hamms so that's it's not just Brad looking at all my instructions related PRs, but cc'ing @islemaster in case he's interested.

### CODE REVIEW

**Comment by wjordan** [comment]
Just a side comment that I really love this approach combining browserify-shim with a common method in the test suite to stub the externals, it should make for a much cleaner (and more testable) JS codebase. Would love to see it eventually extended to jQuery and React down the road. Destroy all globals!

**Comment by islemaster** [comment]
:smile: So much less terrible!

File: apps/src/StudioApp.js [hunk 1]
```diff
@@ -26,6 +26,7 @@ var assetsApi = require('./clientApi').assets;
 var assetPrefix = require('./assetManagement/assetPrefix');
 var assetListStore = require('./assetManagement/assetListStore');
 var annotationList = require('./acemode/annotationList');
+var processMarkup = require('marked');
```
Comment (Hamms) [inline apps/src/StudioApp.js]: nit: shouldn't this be `processMarkdown`?

Comment (Bjvanminnen) [inline apps/src/StudioApp.js]: Yeah, probably. I guess markdown is a particular flavor of markup http://stackoverflow.com/a/24044/2506748, which isn't at all confusing :-P

**Comment by Hamms** [comment]
I love seeing fewer `window` calls scattered around. Destroy all globals!

This looks great to me.

File: apps/src/StudioApp.js (hunk 1 above)
Comment (Hamms) [inline apps/src/StudioApp.js]: That's the problem with clever names.

**Pull request merged**